src/Omega_h_array.hpp

```diff
@@ -47,57 +65,16 @@ class Write {
   Write(LO size_in, T offset, T stride, std::string const& name = "");
   Write(std::initializer_list<T> l, std::string const& name = "");
   Write(HostWrite<T> host_write);
-  OMEGA_H_INLINE LO size() const OMEGA_H_NOEXCEPT {
```

Comment:
dont the array default and arraykokkos headers need to include array.hpp

Reply:
No, it's actually the other way around, array.hpp includes arraydefault or arraykokkos. arraydefault and arraykokkos provide the out-of-line definitions for the templated Write class. Since the definitions differ between the existing backend and the Kokkos backend, I saw it fit to separate these out-of-line definitions into two separate header files to reduce on the preprocessor directives in an effort to make the code more readable. These definitions must be in a header file as Write is templated, and after declaration of the Write class. As such they are included by array.hpp at the end of file since the #include preprocessor directive in effect "copies and pastes" the contents of the file into the spot where the include is, thus satisfying the requirements of out-of-line definitions of a templated class.